File: aliasing_self/aliasing_self.py
class Person:
    def __init__(self, foods):
        self.favorite_foods = foods

    def add_favorites(self, other_person):
        for food in other_person.favorite_foods:
            self.favorite_foods.append(food)

alice = Person(["apples"])
alice.add_favorites(alice)

File: aliasing_self/aliasing_self.rs
struct Person {
    favorite_foods: Vec<&'static str>,
}

impl Person {
    fn add_favorites(&mut self, other: &Person) {
        for food in &other.favorite_foods {
            self.favorite_foods.push(food);
        }
    }
}

fn main() {
    let mut alice = Person {
        favorite_foods: vec!["apples"],
    };
    alice.add_favorites(&alice);
}
